Extension/src/LanguageServer/customProviders.ts

```diff
@@ -1,185 +1,208 @@
+        return this._version;
+    }
+
+    public get name(): string {
+        return this.provider.name;
+    }
+
+    public get extensionId(): string {
+        return this._version === Version.v0 ? this.provider.name : this.provider.extensionId;
+    }
+
+    public canProvideConfiguration(uri: vscode.Uri, token?: vscode.CancellationToken): Thenable<boolean> {
+        return this.provider.canProvideConfiguration(uri, token);
+    }
+
+    public provideConfigurations(uris: vscode.Uri[], token?: vscode.CancellationToken): Thenable<SourceFileConfigurationItem[]> {
+        return this.provider.provideConfigurations(uris, token);
+    }
+
+    public canProvideBrowseConfiguration(token?: vscode.CancellationToken): Thenable<boolean> {
+        return this._version < Version.v2 ? Promise.resolve(false) : this.provider.canProvideBrowseConfiguration(token);
+    }
+
+    public provideBrowseConfiguration(token?: vscode.CancellationToken): Thenable<WorkspaceBrowseConfiguration> {
+        return this._version < Version.v2 ? Promise.resolve({browsePath: []}) : this.provider.provideBrowseConfiguration(token);
+    }
+
+    public dispose(): void {
+        if (this._version !== Version.v0) {
+            this.provider.dispose();
+        }
+    }
+}
+
+export class CustomConfigurationProviderCollection {
+    private providers: Map<string, CustomProviderWrapper> = new Map<string, CustomProviderWrapper>();
+
+    private logProblems(provider: CustomConfigurationProvider, version: Version): void {
+        let missing: string[] = [];
+        if (!provider.name) {
+            missing.push("'name'");
+        }
+        if (version !== Version.v0 && !provider.extensionId) {
+            missing.push("'extensionId'");
+        }
+        if (!provider.canProvideConfiguration) {
+            missing.push("'canProvideConfiguration'");
+        }
+        if (!provider.provideConfigurations) {
+            missing.push("'canProvideConfiguration'");
+        }
+        if (version !== Version.v0 && !provider.dispose) {
+            missing.push("'dispose'");
+        }
+        if (version >= Version.v2 && !provider.canProvideBrowseConfiguration) {
+            missing.push("'canProvideBrowseConfiguration'");
+        }
+        if (version >= Version.v2 && !provider.provideBrowseConfiguration) {
+            missing.push("'provideBrowseConfiguration'");
+        }
+        console.error(`CustomConfigurationProvider was not registered. The following properties are missing from the implementation: ${missing.join(", ")}.`);
+    }
+
+    private getId(provider: string|CustomConfigurationProvider): string {
+        if (typeof provider === "string") {
+            return provider;
+        } else if (provider.extensionId) {
+            return provider.extensionId;
+        } else if (provider.name) {
+            return provider.name;
+        } else {
+            console.error(`invalid provider: ${provider}`);
+            return "";

```

Comment:
@grdowns There's an option to "Hide whitespace changes" which fixes the diff problems in this case.

Reply:
I'm not sure what happened.  I think I had the vscode setting to trim extra whitespace enabled at some point.  I wonder if it changed my line endings...

Reply:
@sean-mcmanus Yep, that does the trick. Since this is the case then yeah @bobbrow it sounds like the line endings were converted

Reply:
Yeah, it looks like the line endings were \n for those files and now they are \r\n.  I had thought that git would normalize everything to \n on the server, but we might need to set that attribute manually in the .gitattributes file.  But we should do that in a separate change.  I'll see if I can revert these files back to \n to reduce the diff noise.

Reply:
It turned out that the line endings thing happened as a result of me installing a private fork of git to support the Windows codebase.  I have since fixed the settings problem so this doesn't happen again.